Tests/AppTests/SitemapControllerTests.swift

```diff
@@ -0,0 +1,55 @@
+// Copyright Dave Verwer, Sven A. Schmidt, and other contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@testable import App
+
+import Plot
+import SnapshotTesting
+import XCTVapor
+
+
+class SitemapControllerTests: SnapshotTestCase {
+
+    @MainActor
+    func test_siteMapIndex() async throws {
+        // Setup
+        let packages = (0..<3).map { Package(url: "\($0)".url) }
+        try await packages.save(on: app.db)
+        try await packages.map { try Repository(package: $0, defaultBranch: "default",
+                                                lastCommitDate: Current.date(), name: $0.url,
+                                                owner: "foo") }.save(on: app.db)
+        try await packages.map { try Version(package: $0, packageName: "foo",
+                                             reference: .branch("default")) }.save(on: app.db)
+        try await Search.refresh(on: app.db).get()
+
+        // MUT
+        let req = Vapor.Request(application: app, url: "/sitemap-static-pages.xml",on: app.eventLoopGroup.next())
+        let response = try await SiteMapController.index(req: req)
+
+        // Validation
+        let body = try XCTUnwrap(response.body.string)
+        assertSnapshot(matching: body, as: .init(pathExtension: "xml", diffing: .lines))
+    }
+
+    @MainActor
+    func test_siteMapStaticPages() async throws {
+        // MUT
+        let req = Vapor.Request(application: app, url: "/sitemap-static-pages.xml",on: app.eventLoopGroup.next())
```

Comment:
@finestructure I switched these for tests similar to what you suggested in https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/pull/2432#discussion_r1230522835. Unfortunately, it's not possible to pretty print as you suggested in that comment, but I think a compressed output works reasonably for sitemaps and this way is better than needing a pair of `siteMap` and `buildSiteMap` methods, as we had when you first reviewed this.

This way, we just have the one method and we test the one method.

Reply:
One thing to note here: these _only_ test the returned content and _not_ whether they're wired up via routes.

I think it would be better if we test both at the same time as we normally do in these cases, via

```
app.test(.GET, "/sitemap-index.xml") { res in
    // assert status and body
}
```

Reply:
I just changed that an I'm actually getting a 404, because `/sitemap-index.xml` isn't routed - it's `/sitemap.xml` according to `SiteURL.siteMapIndex`. I'm not sure which URL it should be, so I'll adjust the test to match what's in `SiteURL` for now.